### PR TITLE
Add missing tools

### PR DESCRIPTION
**Pull request by maciejdrozdowski**
- add Dextr Tool
- add Keypoint Tool ( without keypoint management )
- add Wand Tool
- add magic-wand-tool library and add d.ts with required types for project
- enable scale mechanism

### CODE REVIEW

File: client/src/Annotator/Panel/Tool/Dextr.tsx [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
+    setPadding,
+    setThreshold,
+}) => (
+    <Box className={className}>
+        <TextField
+            label="Threshold"
+            inputProps={{ style: { textAlign: 'center' } }}
+            value={padding}
+            onChange={(e: ChangeEvent<HTMLInputElement>) => {
+                let val = parseInt(e.target.value);
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Panel/Tool/Dextr.tsx]: why is this declared as let?

File: client/src/Annotator/Panel/Tool/Dextr.tsx [hunk 2]
```diff
@@ -1,15 +1,41 @@
...
+                setPadding(val ? val : 1);
+            }}
+        />
+        <TextField
+            label="Blur"
+            inputProps={{ style: { textAlign: 'center' } }}
+            value={threshold}
+            onChange={(e: ChangeEvent<HTMLInputElement>) => {
+                let val = parseInt(e.target.value);
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Panel/Tool/Dextr.tsx]: ditto

File: client/src/Annotator/Panel/Tool/Wand.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    setThreshold,
+    setBlur,
+}) => (
+    <Box className={className}>
+        <TextField
+            label="Threshold"
+            inputProps={{ style: { textAlign: 'center' } }}
+            value={threshold}
+            onChange={(e: ChangeEvent<HTMLInputElement>) => {
+                let val = parseInt(e.target.value);
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Panel/Tool/Wand.tsx]: ditto

File: client/src/Annotator/Panel/Tool/Wand.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                setThreshold(val ? val : 1);
+            }}
+        />
+        <TextField
+            label="Blur"
+            inputProps={{ style: { textAlign: 'center' } }}
+            value={blur}
+            onChange={(e: ChangeEvent<HTMLInputElement>) => {
+                let val = parseInt(e.target.value);
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Panel/Tool/Wand.tsx]: ditto

File: client/src/Annotator/Paper/Tool/useDextr.tsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+            const data = await AnnotatorApi.dextr(
+                imageId,
+                pointsList,
+                settings,
+            );
+            if (data.disabled || !data.segmentaiton) return;
+
+            const cPath = new paper.Path({}); // TODO check if compoundPath
+            for (let i = 0; i < data.segmentaiton.length; i++) {
+                let polygon = data.segmentaiton[i];
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Paper/Tool/useDextr.tsx]: again why is this `let`?

File: client/src/Annotator/Paper/Tool/useDextr.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    useEffect(() => {
+        if (toolRef.current != null && isActive) {
+            toolRef.current.activate();
+        }
+    }, [isActive]);
+
+    return {
+        padding: settings.padding,
+        threshold: settings.threshold,
+        setPadding: setPadding,
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Paper/Tool/useDextr.tsx]: use es6 short syntax

File: client/src/Annotator/Paper/Tool/useWand.tsx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    const setThreshold = useCallback((value: number) => {
+        _setSettings(oldState => ({ ...oldState, threshold: value }));
+    }, []);
+
+    const setBlur = useCallback((value: number) => {
+        _setSettings(oldState => ({ ...oldState, blur: value }));
+    }, []);
+
+    const _flood = useCallback(
+        (
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Paper/Tool/useWand.tsx]: if function takes so many arguments it should take object as an argument to use the named params.

File: client/src/Annotator/Paper/Tool/useWand.tsx [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+            const image = {
+                data: imageData.data,
+                width: width,
+                height: height,
+                bytes: 4,
+            };
+
+            const newRadius = rad < 1 ? 1 : Math.abs(rad); // TODO check if we should save reference
+
+            let mask = MagicWand.floodFill(image, x, y, threshold);
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Paper/Tool/useWand.tsx]: you create a variable only to override it in the next line. Why is that?

File: client/src/Annotator/Paper/Tool/useWand.tsx [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+            );
+            if (contours[0]) {
+                let centerX = width / 2;
+                let centerY = height / 2;
+                let points = contours[0].points;
+                points = points.map(pt => ({
+                    x: pt.x + 0.5 - centerX,
+                    y: pt.y + 0.5 - centerY,
+                }));
+                let polygon = new paper.Path(points);
```
Comment (wojciech-bilicki) [inline client/src/Annotator/Paper/Tool/useWand.tsx]: why is this `let`? WTF

**Review by wojciech-bilicki — CHANGES_REQUESTED** [review]

**Review by wojciech-bilicki — APPROVED** [review]

**Review by wiola-dreszler — APPROVED** [review]

**Pull request merged**